### PR TITLE
Implemented custom prefixes for flags and options

### LINKED ISSUE #19: Custom prefixes for flags
_As a CLI developer, I feel like using only `--` or `-` prefixes for flags or options is not enough, so I need a way to customise prefix of a flag to be like `+` or `/`._

### PR DESCRIPTION
**Pull request by JoshuaLight**
Closes #19.

### CODE REVIEW

File: candies/cli/parsers/standard.py [hunk 1]
```diff
@@ -103,37 +131,53 @@ def configured(new: Callable, command: Command, prefix: str = '.') \
...
+    def short_of(x: inspect.Parameter, kind: Any) -> Optional[str]:
+        short = getattr(kind, 'short', None)
 
         if short is None:
             # If `None`, then considered as omitted.
             return short
 
         if short == '':
-            raise ValueError(f"Flag '{parameter.name}' has invalid short name "
+            raise ValueError(f"Argument '{x.name}' has invalid short name "
```
Comment (konstantin-ogulchansky) [inline candies/cli/parsers/standard.py:148]: There is an article missing: "has _an_ invalid short name".

File: candies/cli/parsers/standard.py [hunk 2]
```diff
@@ -103,37 +131,53 @@ def configured(new: Callable, command: Command, prefix: str = '.') \
...
-        # TODO: Add more prefixes depending on `prefix` attribute.
-        return '-', '--'
+    def prefixes_of(x: inspect.Parameter, kind: Any) -> Tuple[str, str]:
+        prefix = getattr(kind, 'prefix', None)
+
+        if prefix is None:
+            return '-', '--'
+
+        if prefix == '':
+            raise ValueError(f"Argument '{x.name}' has invalid prefix "
```
Comment (konstantin-ogulchansky) [inline candies/cli/parsers/standard.py:171]: There is an article missing: "has an invalid prefix".

File: candies/cli/parsers/standard.py [hunk 3]
```diff
@@ -78,6 +78,34 @@ def format_help(self):
...
+
+    def prefix(x: inspect.Parameter) -> str:
+        if isinstance(x.annotation, type):
+            # This allows writing `x: Flag` and expect `--x` to be parsed.
+            return '-'
+        else:
+            return getattr(x.annotation, 'prefix', None)
+
+    def prefix_is_one_character(x: str) -> bool:
+        return x is not None and len(x) == 1
```
Comment (konstantin-ogulchansky) [inline candies/cli/parsers/standard.py:96]: Do we really need to check for `len(x) == 1`? As far as I understand, if any `len(prefix) > 1`, this method will fail with an exception anyway. If so, I believe we could get rid of `prefix_is_one_character` by calling `getattr(x.annotation, 'prefix', '-')`.

File: candies/cli/args/flag.py [hunk 4]
```diff
@@ -14,9 +14,13 @@ class Flag:
...
 
     Attributes:
-        description (str): A description of a flag. Shown in the help page.
-        short (str): A short form of a flag.
+        description: A description of a flag. Shown in the help page.
+        short: A short form of a flag.
             Must be non empty and consist of 1 alphabet character.
+        prefix: A string that represents a prefix of a flag.
+            It should contain only 1 character.
+            For example, a value of `+` will allow `++flag` and `+f` notation.
```
Comment (JoshuaLight) [inline candies/cli/args/flag.py:22]: Probably `allows` would be better than `will allow`.

File: candies/cli/args/opt.py [hunk 5]
```diff
@@ -16,9 +16,14 @@ class Opt:
     `--times` is an option with value of `3`.
 
     Attributes:
-        description (str): A description of an option. Shown in the help page.
-        short (str): A short form of an option.
+        description: A description of an option. Shown in the help page.
+        short: A short form of an option.
             Must be non empty and consist of 1 alphabet character.
+        prefix: A string that represents a prefix of an option.
```
Comment (JoshuaLight) [inline candies/cli/args/opt.py:22]: Added only description for an argument, not the argument itself.

File: candies/cli/args/flag.py (hunk 4 above)
Comment (JoshuaLight) [reply candies/cli/args/flag.py]: Fixed.

File: candies/cli/args/opt.py (hunk 5 above)
Comment (JoshuaLight) [reply candies/cli/args/opt.py:22]: Fixed. Added test for that.

File: candies/cli/parsers/standard.py (hunk 3 above)
Comment (JoshuaLight) [reply candies/cli/parsers/standard.py]: Agreed. I've refactored code a bit. Could you please take a look?

File: candies/cli/parsers/standard.py (hunk 2 above)
Comment (JoshuaLight) [reply candies/cli/parsers/standard.py]: Fixed.

File: candies/cli/parsers/standard.py (hunk 1 above)
Comment (JoshuaLight) [reply candies/cli/parsers/standard.py]: Fixed.

File: candies/cli/parsers/standard.py [hunk 6]
```diff
@@ -78,6 +78,38 @@ def format_help(self):
...
+        return prefix
+
+    params = signature.parameters.values()
+    prefixes = map(prefix_of, params)
+    prefixes = set(prefixes)
+
+    if len(prefixes) == 0:
+        return '-'
+
+    return ''.join(prefixes)
```
Comment (konstantin-ogulchansky) [inline candies/cli/parsers/standard.py:110]: What do you think about `return ''.join(prefixes) or '-'`?

Comment (JoshuaLight) [reply candies/cli/parsers/standard.py]: Great!

**Pull request merged**